### PR TITLE
feat: use require for polyfills and helpers

### PR DESCRIPTION
**Pull request by RiESAEX**
**Description:**

use require for polyfills and helpers
typeof isModule changed to Option<IsModule>.to allow test case set there isModule with swcrc.
https://github.com/swc-project/swc/discussions/6760
**BREAKING CHANGE:**

**Related issue (if exists):**
close https://github.com/swc-project/swc/issues/6727

### CODE REVIEW

File: crates/swc_ecma_preset_env/src/lib.rs [hunk 1]
```diff
@@ -452,8 +464,64 @@ impl VisitMut for Polyfills {
         m.body.retain(|item| !matches!(item, ModuleItem::ModuleDecl(ModuleDecl::Import(ImportDecl { src, .. })) if src.span == DUMMY_SP && src.value == js_word!("")));
     }
 
-    fn visit_mut_script(&mut self, _: &mut Script) {
-        unimplemented!("automatic polyfill for scripts")
+    fn visit_mut_script(&mut self, m: &mut Script) {
+        let span = m.span;
+        let required = self.collect(m);
+        if cfg!(debug_assertions) {
```
Comment (kdy1) [inline crates/swc_ecma_preset_env/src/lib.rs:470]: What's the purpose of this?

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_preset_env/src/lib.rs (hunk 1 above)
Comment (RiESAEX) [reply crates/swc_ecma_preset_env/src/lib.rs:470]: I don't known, but visit_mut_module is written like this.

File: crates/swc/tests/fixture/issues-6xxx/6418/1/output/input.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
-var foo = function o() {};
+var foo = function foo() {};
```
Comment (RiESAEX) [inline crates/swc/tests/fixture/issues-6xxx/6418/1/output/input.js:1]: I'm not sure about this change..
I think this is caused by the type change of isModule.

**Comment by RiESAEX** [comment]
Wasm build failed, but I don't know how to fix it. It works fine on my computer.

File: crates/swc_ecma_transforms_base/src/helpers/mod.rs [hunk 3]
```diff
@@ -92,6 +91,52 @@ macro_rules! add_import_to {
     }};
 }
 
+macro_rules! add_require_to {
```
Comment (kdy1) [inline crates/swc_ecma_transforms_base/src/helpers/mod.rs:94]: Please change this to a function.
It will fix Wasm issue

**Review by kdy1 — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_transforms_base/src/helpers/mod.rs [hunk 4]
```diff
@@ -391,13 +407,58 @@ impl InjectHelpers {
...
+        };
+        let decl = Decl::Var(
+            VarDecl {
+                span: DUMMY_SP,
+                kind: VarDeclKind::Var,
+                declare: false,
+                decls: vec![VarDeclarator {
+                    span: DUMMY_SP,
+                    name: Pat::Ident(
+                        Ident::new(("_".to_string() + name).into(), DUMMY_SP.apply_mark(mark))
```
Comment (kdy1) [inline crates/swc_ecma_transforms_base/src/helpers/mod.rs:445]:
```suggestion
                        Ident::new(format!("_{}", name).into(), DUMMY_SP.apply_mark(mark))
```

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_transforms_base --breaking

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI failed

**Review by kdy1 — APPROVED** [review]

**Pull request merged**